src/utils/openmc/tallies.py

```diff
@@ -1443,19 +835,61 @@ def __init__(self):
         self._tallies_file = ET.Element("tallies")
 
 
-    def add_tally(self, tally):
+    def add_tally(self, tally, merge=False):
 
         if not isinstance(tally, Tally):
             msg = 'Unable to add a non-Tally {0} to the TalliesFile'.format(tally)
             raise ValueError(msg)
 
-        self._tallies.append(tally)
+        if merge:
+            merged = False
+
+            # Look for a tally to merge with this one
+            for i, tally2 in enumerate(self._tallies):
+
+                # If a mergeable tally is found
+                if tally2.can_merge(tally):
+
+                    # Replace tally 2 with the merged tally
+                    merged_tally = tally2.merge(tally)
+                    self._tallies[i] = merged_tally
+                    merged = True
+
+            # If not mergeable tally was found, simply add this tally
+            if not merged:
+                self._tallies.append(tally)
+
+        else:
+            self._tallies.append(tally)
 
 
     def remove_tally(self, tally):
         self._tallies.remove(tally)
 
 
+    def merge_tallies(self):
+
+        for i, tally1 in enumerate(self._tallies):
+            for j, tally2 in enumerate(self._tallies):
+
+                # Do not merge the same tally with itself
+                if i == j:
+                    continue
```

Comment:
The reason for current loop structure is that it enables ***n*-way** merging of tallies. In other words, if four tallies in the list are *mergeable*, then all four will be merged into a single tally. This capability is broken if the second loop is restructured such that it only considers subsequent tallies - in this case, the first two mergeable tallies would be merged, and the second two would be merged, for a total of two new merged tallies rather than a single one. Unless I'm missing something, I believe my argument here is correct - I've tested this for some of my stuff when I create 1000+ distinct tallies, with *1-8*-way mergeability and it seems to perform as expected.

I do realize that I may be breaking some Pythonic rules here by iterating over loops which change size, but my understanding is that this is fine when using ``enumerate`` as is done here.